### DISCUSSION TITLE
How to do custom sorting in svelte-headless-table?

### DISCUSSION THREAD

**Discussion by Git002** [bryanmylee/svelte-headless-table#198] [Q&A] [answered]
Actually, I have dates in my table and they are in this format: "Wed Mar 25 2015", now I am wondering, if I want to implement sorting according to dates then how to? Also maybe in future I might want a custom sorting according to different data, so how could one achieve this?

```
const table = createTable(data, {
  sort: addSortBy()
});
```

**Comment by Git002** [comment, marked as answer]
Okay I figured it out. It was written in docs too but I wished there was a working example too, but I am greatful for the docs anyways. Leaving this, as it might be helpful to others too...

Here's how I did it for sorting dates well. You will have to modify the logic inside the `compareFn` yourself though according to your needs, the rest of the syntax will be the same.

```
table.column({
    header: tableHeaderKeys[key],
    accessor: key,
    plugins: {
        sort: {
            compareFn(left, right) {
                if (new Date(left) < new Date(right)) return -1;
                else if (new Date(left) > new Date(right)) return 1;
                else return 0;
            }
        }
    }
})
```

**Reply by milespatterson-8451** [reply]
Thanks for the example.

You can also assign the compareFn like so:
```javascript

const dateColCmp = (left, right) => {
    if (new Date(left) < new Date(right)) return -1;
    else if (new Date(left) > new Date(right)) return 1;
    else return 0;
}

table.column({
    header: tableHeaderKeys[key],
    accessor: key,
    plugins: {
        sort: {
            compareFn: dateColCmp
        }
    }
})
```

to keep the column def a little cleaner. We have some pretty complex tab cols and it became extremely hard to read.

Thank you for the example, it was nice to visualize this
